lib/services/TokenService.ts

```diff
@@ -0,0 +1,86 @@
+/* global window */
+import { TokenManager, EVENT_EXPIRED } from '../TokenManager';
+import { AuthSdkError } from '../errors';
+import { isBrowser } from '../features';
+import { TokenManagerOptions } from '../types';
+
+function shouldThrottleRenew(renewTimeQueue) {
+  let res = false;
+  renewTimeQueue.push(Date.now());
+  if (renewTimeQueue.length >= 10) {
+    // get and remove first item from queue
+    const firstTime = renewTimeQueue.shift();
+    const lastTime = renewTimeQueue[renewTimeQueue.length - 1];
+    res = lastTime - firstTime < 30 * 1000;
+  }
+  return res;
+}
+
+export class TokenService {
+  private tokenManager: TokenManager;
```

Comment:
I was thinking should we use `private` in the code, as it only make sense in typescript/esm, but it become `public` after transpilation. Probably we can use both (private access modifier and the `_` convention) for private fields.